interfaces/builtin/fwupd.go

```diff
@@ -65,17 +65,26 @@ const fwupdPermanentSlotAppArmor = `
   # Allow fwupd to access system information
   /sys/devices/virtual/dmi/id/product_name r,
   /sys/devices/virtual/dmi/id/sys_vendor r,
+  /sys/firmware/dmi/tables/ r,
+  /sys/firmware/dmi/tables/** r,
 
   # Allow read/write access for efivarfs filesystem
   /sys/firmware/efi/efivars/ r,
   /sys/firmware/efi/efivars/** rw,
 
   # Allow write access for efi firmware updater
-  /boot/efi/EFI/ubuntu/fw/** rw,
+  /boot/efi/ r,
+  /boot/efi/** r,
+  /boot/efi/EFI/ubuntu/** rw,
```

Comment:
This allows rewriting grub, mok-manager, BOOTX64.CSV, etc. What are do you actually need to update, fwupdx64.efi? If that, how about adjusting these rules to be:
```
/boot/efi/{,**/} r,
/boot/efi/EFI/fwupd*.efi* rw,
/boot/efi/EFI/ubuntu/fw/** rw,
```

This allows reading all the directories in `/boot/efi` as well as rw access to `fwupd*.efi*` and `/boot/efi/EFI/ubuntu/fw/*`. If this doesn't work for you, please paste the policy denials you are seeing.

Reply:
Indeed, fwupd actually only pushes fwupd*.efi under /boot/efi/EFI/ubuntu (not /boot/efi/EFI/), so changing to "/boot/efi/EFI/ubuntu/fwupd*.efi* rw" is reasonable. I'll update these according to suggestions later.